### PR TITLE
remove dependencies to graph

### LINKED ISSUE #44: decouple object storage backends from graph backend
move needed resources to local database

---
addresses https://github.com/SwissDataScienceCenter/renga/issues/135 https://github.com/SwissDataScienceCenter/renga/issues/146

### LINKED ISSUE #50: update swagger
keep swagger definition up to date with the new API

---
addresses https://github.com/SwissDataScienceCenter/renga/issues/135 https://github.com/SwissDataScienceCenter/renga/issues/146

### LINKED ISSUE #51: LFS object backend / bucket choice
by default de-duplication applies, but we may want to limit it per backend

---
addresses https://github.com/SwissDataScienceCenter/renga/issues/135 https://github.com/SwissDataScienceCenter/renga/issues/146

### PR DESCRIPTION
**Pull request by ebiiii**
Fix #44
Fix #51 
Fix #50

### CODE REVIEW

File: app/controllers/ObjectController.scala [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        ) yield {
+          orchestrator.repositories.findByUUID( repo_id ).flatMap( f => {
+            val upload = for (
+              repo <- f;
+              back <- backends.getBackend( repo.backend )
+            ) yield {
+              val filename = reqh.headers.get( "Content-Filename" ).getOrElse( oid )
+              val fo = FileObject( obj_id, "", filename, reqh.headers.get( "Content-Hash" ).getOrElse( "" ), Some( Instant.now() ), UUID.fromString( profile.getId ) )
+              val fr = FileObjectRepository( obj_id, repo.uuid, Some( filename + now.toString ), Some( Instant.now() ) )
+              for (
```
Comment (leafty) [inline app/controllers/ObjectController.scala]: transactionally: http://slick.lightbend.com/doc/3.2.1/dbio.html

File: app/controllers/GitController.scala [hunk 2]
```diff
@@ -52,29 +37,23 @@ import play.api.libs.ws._
...
-    implicit val janusGraphTraversalSourceProvider: JanusGraphTraversalSourceProvider,
-    implicit val vertexReader:                      VertexReader
-) extends Controller with ControllerWithBodyParseTolerantJson with ControllerWithGraphTraversal with RequestHelper {
-
-  lazy val gc: GraphMutationClient = graphMutationClientProvider.get
+    config:                     play.api.Configuration,
+    jwtVerifier:                JWTVerifierProvider,
+    backends:                   Backends,
+    implicit val wsclient:      WSClient,
+    protected val orchestrator: DatabaseLayer
```
Comment (leafty) [inline app/controllers/GitController.scala]: suggestion: rename to `databaseAccessLayer`

File: app/controllers/GitController.scala [hunk 3]
```diff
@@ -144,137 +123,49 @@ class GitController @Inject() (
...
-          else
-            Future( None )
+    val json = JsString( id )
+    json.validate[UUID] match {
+      case JsError( e ) => Future( BadRequest( JsError.toJson( e ) ) )
+      case JsSuccess( uuid, _ ) =>
+        if ( request.body.operation == "download" ) {
+          val objects = request.body.objects.map( lfsObject => {
+            orchestrator.fileobjects.findByHash( lfsObject.oid ).flatMap( _.map( fo =>
+              orchestrator.fileobjectrepositories.listByFileObject( fo.uuid ).map( _.headOption.map( rep =>
```
Comment (leafty) [inline app/controllers/GitController.scala]: useCamelCase

File: app/models/persistence/FileObjectComponent.scala [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    def all(): Future[Seq[FileObject]] = {
+
+      db.run( fileobjects.result )
+    }
+    def insert( r: FileObject ): Future[Int] = {
+      db.run( fileobjects += r )
+    }
+    def update( r: FileObject ): Future[Int] = {
+      db.run( ( for { c <- fileobjects if c.uuid === r.uuid } yield c ).update( r ) )
```
Comment (leafty) [inline app/models/persistence/FileObjectComponent.scala]: TODO: remove `db.run`

File: app/controllers/ObjectController.scala (hunk 1 above)
Comment (ebiiii) [reply app/controllers/ObjectController.scala]: done

File: app/controllers/GitController.scala (hunk 2 above)
Comment (ebiiii) [reply app/controllers/GitController.scala]: used "dal"

File: app/controllers/GitController.scala (hunk 3 above)
Comment (ebiiii) [reply app/controllers/GitController.scala]: done

File: app/models/persistence/FileObjectComponent.scala (hunk 4 above)
Comment (ebiiii) [reply app/models/persistence/FileObjectComponent.scala]: done

File: app/controllers/GitController.scala [hunk 5]
```diff
@@ -131,23 +134,23 @@ class GitController @Inject() (
       case JsSuccess( uuid, _ ) =>
         if ( request.body.operation == "download" ) {
           val objects = request.body.objects.map( lfsObject => {
-            dal.fileObjects.findByHash( lfsObject.oid ).flatMap( _.map(fo =>
-              dal.fileObjectRepositories.listByFileObject( fo.uuid ).map( _.headOption.map(rep =>
+            db.run( dal.fileObjects.findByHash( lfsObject.oid ) ).flatMap( _.map( fo =>
+              db.run( dal.fileObjectRepositories.listByFileObject( fo.uuid ) ).map( _.headOption.map( rep =>
```
Comment (leafty) [inline app/controllers/GitController.scala]: Am I missing something, or this should be a join?

File: app/models/persistence/DatabaseLayer.scala [hunk 6]
```diff
@@ -45,23 +44,22 @@ class DatabaseLayer @Inject() (
...
-  Await.ready( runner, Duration.Inf )
-  runner.onFailure {
-    case NonFatal( t ) => {
+  val tables = List(
+    TableQuery[Repositories],
+    TableQuery[FileObjects],
+    TableQuery[FileObjectRepositories]
+  )
+
+  val f = db.run( MTable.getTables ).flatMap( v => {
```
Comment (leafty) [inline app/models/persistence/DatabaseLayer.scala]: `f` -> `ready` or `init` or `initializer`?

**Pull request merged**